src/main/java/edu/ucla/library/prl/harvester/OaipmhUtils.java

```diff
@@ -116,12 +112,12 @@ public static Future<List<Set>> listSets(final Vertx aVertx, final URL aBaseURL,
      * @param aFrom The optional timestamp of the last successful run
      * @param aTimeout The value to use for the HTTP timeout
      * @param aUserAgent The value to use for the User-Agent HTTP request header
-     * @return A stream of OAI-PMH records
+     * @return An iterator on OAI-PMH records
      */
-    public static Future<Stream<Record>> listRecords(final Vertx aVertx, final URL aBaseURL, final List<String> aSets,
+    public static Future<Iterator<Record>> listRecords(final Vertx aVertx, final URL aBaseURL, final List<String> aSets,
```

Comment:
Out of curiosity: Why the switch from Stream to Iterator? Processing efficiency? Coding elegance? Memory requirements?

Reply:
I didn't get to the bottom of why this happened, but I was sometimes seeing an NPE when I got an iterator back from the stream and then called `next()` on it (in `updateSolrInBatches`). And then I realized it was silly to transform the iterator into a stream here if we were going to transform it back into an iterator in that method.

Reply:
Makes sense.